### ISSUE TITLE
FixedPoints: sampling with a single point also resamples y

### ISSUE THREAD

**Issue by gunthergl** [rusty1s/pytorch_geometric#1090]
## 🐛 Bug

FixedPoints resamples also y if the pos-matrix contains only a single point. 

It might be that this is intended behaviour - then don't worry just close this issue. I can (and probably should) take care of samples with only one point anyways before learning

## To Reproduce

```
import torch
from torch_geometric.data.data import Data
from torch_geometric.transforms.fixed_points import FixedPoints

pointcloud = Data(pos=torch.randn(100, 3), y=torch.tensor([1]))
single_point = Data(pos=torch.randn(1, 3), y=torch.tensor([1]))

random_select_n_points = FixedPoints(num=10)

print(random_select_n_points(pointcloud))
    # Data(pos=[10, 3], y=[1])
print(random_select_n_points(single_point))
    # Data(pos=[10, 3], y=[10])
```

## Expected behavior

The second example should leave `y=[1]` and should not duplicate the values in y to the amount of fixed points.
```
print(random_select_n_points(single_point))
```
`>>> Data(pos=[10, 3], y=[1])`

## Why does this happen and possible solution: 
In [fixed_points.py @ master](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/transforms/fixed_points.py), the following loop checks if the item is a tensor and if the item.size() is equal to the number of nodes. 

After the number of nodes is in the given scenario equal to one, both conditions hold and y is subset by `choice`
```
for key, item in data:
    if bool(re.search('edge', key)):
        continue

    if torch.is_tensor(item) and item.size(0) == num_nodes:
        data[key] = item[choice]
```

An easy solution could be: 
```
for key, item in data:
    if bool(re.search('edge', key)):
        continue

    if torch.is_tensor(item) and item.size(0) == num_nodes and item.size(0) != 1:
        data[key] = item[choice]
```
Which solves the described problem, but I do not know if there are any side-effects :(

## Environment

* OS: Win10
* Python version: 3.7.0
* PyTorch version: 1.4.0

**Comment by rusty1s** [comment]
You are right. That does not make sense. Fixed in master!

**Comment by gunthergl** [comment]
I just reinstalled torch_geometric 1.5.0 
The bug is still reproducible here? I saw 61865eeddc22b0dfbb310c9816622da7f01403f5 which is also the code I now have.

**Comment by gunthergl** [comment]
@rusty1s  could that go into a release? Current version 1.6.1 still has this problem.

**Comment by rusty1s** [comment]
Hey, can you clarify the issue once again? What's the problem with `item.size(0) == 1` in this case? How about you use `replace = False`?

**Comment by gunthergl** [comment]
Hi, so atleast for my usecase, `len(Data.y)` is usually independent of `Data.pos.size(0)`. If you disagree here, this is not an issue of pytorch_geometric and I have to handle this usecase by myself. 

If you agree, which seems to be implemented: You subsample only those items which are equal to `num_nodes`. 
This is usually no problem as `len(Data.y) == 1` and `num_nodes = Data.pos.size(0)`. Subsampling (with replacement) only happens on the `Data.some_kind_of_node_values.size(0) == num_nodes`. 
But if  `len(Data.y) == 1` *and* `num_nodes=1`, `Data.y` is regarded as a node-like value and subsampled. 

This issue might also be if `len(Data.y) == num_nodes` where `num_nodes != 1` but I do not know if there is a practical usecase for this. 

So, `replace=False` would circumvent the problem but I **want** replacement with the possibility that the number of nodes **and** the length of `y` is 1. 

Your fix 61865eeddc22b0dfbb310c9816622da7f01403f5 used a different way than I proposed but I did not check it.
